### ISSUE TITLE
NullPointerException in AbstractLocalAwareExecutorService

### ISSUE THREAD

**Issue by jjengla** [Stratio/cassandra-lucene-index#310]
Cassandra 3.9 with cassandra-lucene-index 3.9.6.
I started getting many of these errors recently and I'm not sure of the cause.  I upgraded from 3.9.3 to 3.9.6 but still get the same errors.  I suspect these are causing repair operations to hang and also seem to correlate with internal timeouts I'm seeing.  Sometimes it shows up with a stack trace, but more recently I just see the Exception.  Any ideas?

```
WARN  [ReadStage-1] 2017-04-05 14:09:09,118 AbstractLocalAwareExecutorService.java:169 - Uncaught exception on thread Thread[ReadStage-1,5,main]: {}
java.lang.NullPointerException: null
        at org.apache.cassandra.db.marshal.CompositeType$Builder.add(CompositeType.java:439) ~[apache-cassandra-3.9.jar:3.9]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.$anonfun$byteBuffer$1(KeyMapper.scala:56) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.mutable.ArrayOps$ofRef.foldLeft(ArrayOps.scala:193) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.mutable.ArrayOps$ofRef.$div$colon(ArrayOps.scala:193) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.byteBuffer(KeyMapper.scala:56) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.bytesRef(KeyMapper.scala:80) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.term(KeyMapper.scala:76) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.query(KeyMapper.scala:90) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.$anonfun$query$1(KeyMapper.scala:101) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.TraversableOnce.$anonfun$foldLeft$1(TraversableOnce.scala:157) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.TraversableOnce.$anonfun$foldLeft$1$adapted(TraversableOnce.scala:157) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.AbstractIterator.foreach(Iterator.scala:932) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.TraversableOnce.foldLeft(TraversableOnce.scala:157) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.TraversableOnce.foldLeft$(TraversableOnce.scala:155) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at scala.collection.AbstractTraversable.$div$colon(Traversable.scala:104) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.mapping.KeyMapper.query(KeyMapper.scala:101) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.IndexServiceWide.query(IndexServiceWide.scala:96) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.IndexService.query(IndexService.scala:302) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.IndexService.search(IndexService.scala:275) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at com.stratio.cassandra.lucene.Index.$anonfun$searcherFor$1(Index.scala:313) ~[cassandra-lucene-index-plugin-3.9.6.jar:na]
        at org.apache.cassandra.db.ReadCommand.executeLocally(ReadCommand.java:398) ~[apache-cassandra-3.9.jar:3.9]
        at org.apache.cassandra.db.ReadCommandVerbHandler.doVerb(ReadCommandVerbHandler.java:48) ~[apache-cassandra-3.9.jar:3.9]
        at org.apache.cassandra.net.MessageDeliveryTask.run(MessageDeliveryTask.java:64) ~[apache-cassandra-3.9.jar:3.9]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_73]
        at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$FutureTask.run(AbstractLocalAwareExecutorService.java:164) ~[apache-cass
andra-3.9.jar:3.9]
        at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$LocalSessionFutureTask.run(AbstractLocalAwareExecutorService.java:136) [
apache-cassandra-3.9.jar:3.9]
        at org.apache.cassandra.concurrent.SEPWorker.run(SEPWorker.java:109) [apache-cassandra-3.9.jar:3.9]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_73]


WARN  [ReadStage-1] 2017-04-06 00:29:51,359 AbstractLocalAwareExecutorService.java:169 - Uncaught exception on thread Thread[ReadStage-1,5,main]: {}
java.lang.NullPointerException: null
WARN  [ReadStage-3] 2017-04-06 00:30:01,392 AbstractLocalAwareExecutorService.java:169 - Uncaught exception on thread Thread[ReadStage-3,5,main]: {}
java.lang.NullPointerException: null
WARN  [ReadStage-4] 2017-04-06 00:30:10,715 AbstractLocalAwareExecutorService.java:169 - Uncaught exception on thread Thread[ReadStage-4,5,main]: {}
java.lang.NullPointerException: null
```
-JE

**Comment by ealonsodb** [comment]
Hello @jjengla 

Based on your logs you are executing a lucene query. Can you please show us the complete CQL example that produces this issue?

You have upgraded from 3.9.3 to 3.9.6. How have you done it? Have you recreated the indexes? Have you followed the [upgrade instructions](https://github.com/Stratio/cassandra-lucene-index/blob/branch-3.9/doc/documentation.rst#upgrade)?

Regards

**Comment by jjengla** [comment]
I did delete and recreate the index, although it looks like I didn't need to.  These logs are all from the cassandra system.log, so I don't know the exact queries being done on the client that trigger them.  Is there any other information I can try to get to debug this?

**Comment by ealonsodb** [comment]
Hi @jjengla: 

Don't you know every query done in your system? 
Don't you know to execute every query through CQLSH while looking system.log?

Based on logs the query is a lucene query against a wide rows table.

Yesterday we released [3.9.7](https://github.com/Stratio/cassandra-lucene-index/tree/3.9.7). 

Can you test if this issue still happens in that version?

Hope this helps
